posthog/hogql_queries/insights/retention_appearance_query_runner.py

```diff
@@ -0,0 +1,140 @@
+                        length(appearances) DESC,
+                        actor_id
+                """,
+                placeholders,
+                timings=self.timings,
+            )
+        return retention_query
+
+    def to_persons_query(self) -> ast.SelectQuery:
+        appearances_query = self.to_query()
+        # Only select actor_id for the persons query
+        appearances_query.select = [ast.Field(chain=["actor_id"])]
+        appearances_query.order_by = []
+        return appearances_query
+
+    def get_actors_from_result(self, raw_result) -> Union[List[SerializedGroup], List[SerializedPerson]]:
+        # TODO: Overhaul by leveraging common functionality in PersonsQueryRunner
+
+        serialized_actors: Union[List[SerializedGroup], List[SerializedPerson]]
+
+        actor_ids = [row[0] for row in raw_result]
+        value_per_actor_id = None
+
+        if self.query.source.aggregation_group_type_index is not None:
+            _, serialized_actors = get_groups(
+                self.team.pk,
+                self.query.source.aggregation_group_type_index,
+                actor_ids,
+                value_per_actor_id,
+            )
+        else:
+            _, serialized_actors = get_people(self.team, actor_ids, value_per_actor_id)
+
+        return serialized_actors
+
+    def calculate(self) -> PersonsQueryResponse:
+        response = self.paginator.execute_hogql_query(
+            query_type="RetentionQuery",
+            query=self.to_query(),
+            team=self.team,
+            timings=self.timings,
+            modifiers=self.modifiers,
+        )
+        actor_appearances = [
+            {"actor_id": str(row[0]), "appearance_count": len(row[1]), "appearances": row[1]}
+            for row in self.paginator.results
+        ]
+
+        serialized_actors = self.get_actors_from_result(
+            [(actor_appearance["actor_id"],) for actor_appearance in actor_appearances]
+        )
+
+        actors_lookup = {str(actor["id"]): actor for actor in serialized_actors}
+
+        total_intervals = self.query.source.retentionFilter.total_intervals or DEFAULT_TOTAL_INTERVALS
+        actors = [
+            [
+                actors_lookup[actor["actor_id"]],
+                [
+                    1 if interval_number in actor["appearances"] else 0
+                    for interval_number in range(total_intervals - (self.query.selectedInterval or 0))
+                ],
+            ]
+            for actor in actor_appearances
+            if actor["actor_id"] in actors_lookup
+        ]
+        return PersonsQueryResponse(
+            results=actors,
+            timings=response.timings,
+            types=[t for _, t in response.types] if response.types else None,
+            columns=["person", "appearances"],
+            hogql=response.hogql,
+            **self.paginator.response_kwargs(),
+        )
+        #   "missing_persons": len(actor_appearances) - len(actors),
```

Comment:
@mariusandra I guess this number is still important? Would it be alright to have a field on `PersonsQueryResponse` with `extended_data` or something?

Reply:
I honestly don't know the context behind this. It was probably something to do with us getting tired of "why is my data off by a few" messages by users. I don't know how used it is, but if we could have it somehow, it would be great.

Reply:
I added it as field to persons response, since when persons are returned we potentially could always miss some I guess.